plugins/gestures.koplugin/main.lua

```diff
@@ -0,0 +1,1039 @@
+            -- If not, the user has changed them: let him deal with setting new ones if needed.
+            if is_rtl then
+                if util.tableEquals(self.gestures[k], self.defaults[k]) then
+                    self.gestures[k] = self.defaults[v]
+                    reset = true
+                end
+                if util.tableEquals(self.gestures[v], self.defaults[v]) then
+                    self.gestures[v] = self.defaults[k]
+                    reset = true
+                end
+            else
+                if util.tableEquals(self.gestures[k], self.defaults[v]) then
+                    self.gestures[k] = self.defaults[k]
+                    reset = true
+                end
+                if util.tableEquals(self.gestures[v], self.defaults[k]) then
+                    self.gestures[v] = self.defaults[v]
+                    reset = true
+                end
+            end
+        end
+        if reset then
+            logger.info("UI language direction changed: resetting some gestures to direction default")
+        end
+        G_reader_settings:flipNilOrFalse(ges_dir_setting)
+    end
+
+    self.ui.menu:registerToMainMenu(self)
+    self:initGesture()
+end
+
+local gestureTextFunc = function(location, ges)
+    local item = location[ges]
+    local action_name = _("Pass through")
+    if item then
+        local sub_item = next(item)
+        if sub_item == nil then return _("Nothing") end
+        action_name = Dispatcher:getNameFromItem(sub_item)
+        if next(item, sub_item) ~= nil then
+            action_name = _("Many")
+        end
+    end
+    return action_name
+end
+
+function Gestures:gestureTitleFunc(ges)
+    local title = gestures_list[ges] or self:friendlyMultiswipeName(ges)
+    return T(_("%1   (%2)"), title, gestureTextFunc(self.gestures, ges))
+end
+
+function Gestures:genMenu(ges)
+    local sub_items = {}
+    if gestures_list[ges] ~= nil then
+        table.insert(sub_items, {
+            text = T(_("%1 (default)"), gestureTextFunc(self.defaults, ges)),
+            keep_menu_open = true,
+            separator = true,
+            checked_func = function()
+                return util.tableEquals(self.gestures[ges], self.defaults[ges])
+            end,
+            callback = function()
+                self.gestures[ges] = util.tableDeepCopy(self.defaults[ges])
+            end,
+        })
+    end
+    table.insert(sub_items, {
+        text = _("Pass through"),
+        keep_menu_open = true,
+        separator = true,
+        checked_func = function()
+            return self.gestures[ges] == nil
+        end,
+        callback = function()
+            self.gestures[ges] = nil
+        end,
+    })
+    Dispatcher:addSubMenu(sub_items, self.gestures, ges)
+    return sub_items
+end
+
+function Gestures:genSubItem(ges, separator, hold_callback)
+    local reader_only = {tap_top_left_corner=true, hold_top_left_corner=true}
```

Comment:
Is that different from previous code?

Reply:
no, they were/are disabled in FM. (I guess as not to interfere with the home & plus button)